src/builder/PaycallWrapper.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.23;
+
+import {CodeJarHelper} from "./CodeJarHelper.sol";
+import {IQuarkWallet} from "quark-core/src/interfaces/IQuarkWallet.sol";
+import {Paycall} from "../Paycall.sol";
+
+// Helper library to wrap a QuarkOperation from Actions.sol for a Paycall
+library PaycallWrapper {
+    function wrap(IQuarkWallet.QuarkOperation memory operation, uint256 chainId, uint256 maxPaymentCost)
+        internal
+        pure
+        returns (IQuarkWallet.QuarkOperation memory)
+    {
+        bytes[] memory scriptSources = new bytes[](operation.scriptSources.length + 1);
+        for (uint256 i = 0; i < operation.scriptSources.length; i++) {
+            scriptSources[i] = operation.scriptSources[i];
+        }
+        scriptSources[operation.scriptSources.length] = type(Paycall).creationCode;
+
+        return IQuarkWallet.QuarkOperation({
+            nonce: operation.nonce,
+            scriptAddress: CodeJarHelper.getCodeAddress(chainId, type(Paycall).creationCode),
```

Comment:
Paycall has a constructor, so you actually need to pass in something like:

`abi.encodePacked(type(Paycall).creationCode, abi.encode(ETH_USD_PRICE_FEED, USDC));`

Reply:
You may want to consider a library for PriceFeeds to get this information.

Reply:
Synced offline and created `PaymentToken.sol` to store the info we need on-chain for addresses.